File: model/Animal.go
package model

import "github.com/jinzhu/gorm"

type Animal struct {
	gorm.Model `swaggerignore:"true"`
	Name string `json:"name"`
	AnimalType string `json:"animaltype"`
	Race string `json:"race"`
	ProfilePicture string `json:"profilePicture"`
	Picture string `json:"picture"`
	UserID uint `gorm:"TYPE:integer REFERENCES users"`
}

File: lib/models/animal.dart

class Animal{
  int id;
  int user_id;
  String name;
  String animaltype;
  String race;

  //List<Image> picture;

 Animal({this.id, this.user_id, this.name, this.animaltype, this.race});

    factory Animal.fromJson(Map<String,dynamic> json){
        return Animal(
          id : json["ID"],
          user_id : json["UserID"],
          name : json["name"],
          animaltype : json["animaltype"],
          race: json["race"]
         
        );


    }
  
}